### ISSUE TITLE
PipelineParameter default values in DatabricksStep

### ISSUE THREAD

**Issue by jarandaf** [Azure/MachineLearningNotebooks#1220]
It seems only string types are supported. int/float types throw the following error:

```
.../lib/python3.7/site-packages/azureml/pipeline/core/_databricks_step_base.py", line 569, in _encode_string_params
    final_params_list.append(value.replace("|", "|-"))
AttributeError: 'int' object has no attribute 'replace'
```

I don't see this behavior when using pipeline parameters in `PythonScriptStep`.

**Comment by balapv** [comment]
The error says that you cannot do a replace on an int object - which means that the param you are adding is an int.
Can you please provide some more details of what you are trying to do here?

**Comment by jarandaf** [comment]
From what I can see, `DatabricksStep` arguments validation is different than the one performed by a `PythonScriptStep`. The former does not support any other type than `str` for a PipelineParameter default value while the latter does. Can anyone please confirm? Is this the expected behavior?

**Comment by lostmygithubaccount** [comment]
that is not expected behavior and should be considered a bug 

(it looks as simple as casting value to str in the initial code snippet)

**Comment by jarandaf** [comment]
Yeah, that's what I do as a workaround. Moreover I think that other basic types as list should also be supported (right know they aren't).

**Comment by shbijlan** [comment]
@jarandaf Thank you for bringing this to our attention. I have opened a work item to address this change. Once approved by our engineering team, it will be taken up in our planning and release cycle.

**Comment by lostmygithubaccount** [comment]
any updates on this?

**Comment by shbijlan** [comment]
This bug is approved by engineering. I will update as soon as the change is completed and deployed.

**Comment by Di-Tez** [comment]
MSFT internal use: https://msdata.visualstudio.com/Vienna/_queries/edit/959289
